controllers/storagecluster/cephcluster.go

```diff
@@ -445,6 +445,7 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, kmsConfigMap *co
 				Interval: &metav1.Duration{Duration: 30 * time.Second},
 			},
 			Storage: rookCephv1.StorageScopeSpec{
+				AllowDeviceClassUpdate:       true,
```

Comment:
If all went well with #2615 and #2694 to properly set the correct device classes on the OSDs and pools, this will be a no-op and no changes will be needed to the OSD device classes. So I'm having second thoughts about always setting this to true. If there are any bugs in the device classes of the OSDs, it could cause unexpected data movement or pools to break. Instead, I am thinking we should set this to `false` by default and expose a storageCluster setting that allows the admin to set it to true if they do actually want to force update the OSD device class. I would expect only support would need to temporarily enable that setting if they do find an issue with the device classes.

Reply:
Hm, That's actually true the bug attached to 2615 has been verified after testing but 2694 is still under testing, should we wait till 2694 is also tested & verified?

Reply:
Ack, have updated the PR now.